### DISCUSSION TITLE
Z-axis positioning effect of Loco Pos

### DISCUSSION THREAD

**Discussion by liu-hao000** [bitcraze/discussions#737] [Q&A]
When controlling Crazyflies using the LPS system in TDoA3 mode with 8 anchors positioned at x=3m, y=6m, z=2.5m, I observed a significant drop in Z-axis height when the Crazyflie flies to the right side of the video (as seen in this [video](https://www.youtube.com/watch?v=JW44xxDRhyM)), while my code maintains the Z-axis at 1 meter. OptiTrack testing did not exhibit similar issues. What could be causing this problem with my LPS system configuration?

Thank you for your assistance.

**Comment by knmcguire** [comment]
Hi!  For your setup, the differences between anchors in height is not as much as the side and the width, which means that in general there tends to be more uncertainty in the height estimate, but that should be noticable through the entire setup. In your situation, it could be that the right anchors are not mounted in the same direction as the left anchors, as the direction on which the antenna faces can have a huge impact (asymetrical antenna lobes). There could also be that there is some metals or other elements that can reflect the ranges more on one side of the flight area which might distort it as well.

Technically, looking at the video, it looks to be in line with what we would expect from a small setup like this, but you can try perhaps the experimental 'robusttdoa' parameter (from this [blogpost](https://www.bitcraze.io/2020/04/learning-based-bias-correction-for-accurate-ultra-wideband-localization-of-a-crazyflie/))? Maybe it will improve it?

**Reply by liu-hao000** [reply]
Thank you for your reply, I have checked the left and right anchors, but the Z-axis positioning effect is still not very good. If I add two anchors at (0,3,2.5) and (3,3,2.5), theoretically, the Z-axis positioning effect will be better?

**Reply by krichardsson** [reply]
> If I add two anchors at (0,3,2.5) and (3,3,2.5), theoretically, the Z-axis positioning effect will be better?

One more anchor high and low on the problematic side might help. If you have more anchors it is worth trying out.
